contrib/parseq-lambda-names/src/test/java/com/linkedin/parseq/lambda/TestInterface.java

```diff
@@ -0,0 +1,100 @@
+package com.linkedin.parseq.lambda;
+
+import java.util.Optional;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+
+public class TestInterface extends BaseTest {
+
+  private static final String CLASSNAME = TestInterface.class.getSimpleName();
+
+  public interface SampleInterface {
+
+    default String defaultFunction(String s) {
+      return s;
+    }
+
+    static String staticFunction(String s) {
+      return s;
+    }
+
+    String interfaceFunction(String s);
+  }
+
+  public abstract class SampleAbstract {
+
+    abstract String abstractFunction(String s);
+
+  }
+
+  public class SampleImplementation extends SampleAbstract implements SampleInterface {
+
+    @Override
+    String abstractFunction(String s) {
+      return s;
+    }
+
+    @Override
+    public String interfaceFunction(String s) {
+      return s;
+    }
+  }
+
+  @Test
+  public void testFunctionReferenceOnInterface() {
+    SampleImplementation impl = new SampleImplementation();
+    Optional<String> description = getDescriptionForFunction(impl::interfaceFunction);
+    assertTrue(description.isPresent());
+    assertNameMatch("impl::interfaceFunction", "testFunctionReferenceOnInterface", CLASSNAME, description.get().toString());
+  }
+
+  @Test
+  public void testStaticFunctionReferenceOnInterface() {
+    Optional<String> description = getDescriptionForFunction(SampleInterface::staticFunction);
+    assertTrue(description.isPresent());
+    assertNameMatch("TestInterface$SampleInterface::staticFunction", "testStaticFunctionReferenceOnInterface", CLASSNAME, description.get().toString());
+  }
+
+  @Test
+  public void testAbstractFunctionReferenceOnInterface() {
+    SampleImplementation impl = new SampleImplementation();
+    Optional<String> description = getDescriptionForFunction(impl::abstractFunction);
+    assertTrue(description.isPresent());
+    assertNameMatch("impl::abstractFunction", "testAbstractFunctionReferenceOnInterface", CLASSNAME, description.get().toString());
+  }
+
+  @Test
+  public void testFunctionInvocationOnInterface() {
+    SampleImplementation impl = new SampleImplementation();
+    Optional<String> description = getDescriptionForFunction(s -> impl.interfaceFunction(s));
+    assertTrue(description.isPresent());
+    if (JDK_VERSION.equals(JDK_1_8_0_72)) {
+      assertNameMatch("(impl,s) -> interfaceFunction(_)", "testFunctionInvocationOnInterface", CLASSNAME, description.get().toString());
```

Comment:
This doesn't seam right. "s -> interfaceFunction(_)" looks acceptable to me but "(impl,s) -> interfaceFunction(_)" suggests that it is a lambda shape that takes two arguments while it takes only one argument. This might be confusing if one would like to locate it in a source code.

If it is not possible to distinguish between the two then perhaps it would be better to completely ditch left hand side for now. We can always try to improve it in the future versions.
If there is a significant number of cases for which left hand side is reliable then we could replace left hand side where we are unsure with "(?) ->". If it is difficult to even detect where we are unsure then we could remove left hand side completely for now and revisit it in the future.